raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -90,123 +87,122 @@ def test_connect_peers():
             node.stop()
 
 
-@pytest.mark.parametrize("amount_of_nodes", [1])
-def test_subscribe_to_invalid(comms_nodes):
-    client = comms_nodes[1].client
+@pytest.mark.parametrize("nodes_to_clients", [{"A": 1}, {"A": 2}])
+def test_subscribe_to_invalid(comms_clients):
     unregistered_address = generate_address()
+    for client in comms_clients.values():
+        # no subscriptions should be present
+        assert client._is_subscribed_to(unregistered_address) is False
+        # subscribe to unregistered address
+        topic_id, _ = client.subscribe_to(unregistered_address)
+        # FIXME: this should probably throw an exception rather than an empty result
+        assert topic_id is ""
+        assert client._is_subscribed_to(unregistered_address) is False
 
-    # no subscriptions should be present
-    assert client._is_subscribed_to(unregistered_address) is False
-
-    # subscribe to unregistered address
-    topic_id, _ = client.subscribe_to(unregistered_address)
 
-    # check subscription again
-    # FIXME: this should probably throw an exception rather than an empty result
-    assert topic_id is ""
-    assert client._is_subscribed_to(unregistered_address) is False
-
-
-@pytest.mark.parametrize("amount_of_nodes", [1])
-def test_subscribe_to_self(comms_nodes):
-    client, address = comms_nodes[1].client, comms_nodes[1].address
+@pytest.mark.parametrize("nodes_to_clients", [{"A": 1, "B": 2}])
```

Comment:
this is not correct, it should be something like:

`[{"A": 1, "B": 1}, {"A": 2}])`

Reply:
agree, changed